lib/morandi/image_processor.rb

```diff
@@ -228,10 +229,37 @@ def apply_decorations!
       @pb = op.call(nil, @pb)
     end
 
+    def apply_shrink_to_fit!
+      return unless negative_crop?
+
+      op = Morandi::ShrinkToFit.new_from_hash(
+        'crop' => options['crop'],
+        'print_size' => [@width, @height],
+        'shrink' => true
+      )
+
+      @pb = op.call(nil, @pb)
+    end
+
     private
 
     def not_equal_to_one(float)
       (float - 1.0) >= Float::EPSILON
     end
+
+    def unpack_crop_options
+      if options['crop'].is_a?(String) && options['crop'] =~ /^\d+,\d+,\d+,\d+/
+        options['crop'].split(/,/).map(&:to_i)
```

Comment:
I'm surprised Rubocop wasn't all like "there's no point this being a RegExp, it can be a String".